### PR TITLE
[clang-tidy] Simplify RenamerClangTidyCheck API

### PR DESCRIPTION
**Pull request by revane**
Some functions allow a null SourceManager, no SourceManager, or a SourceManager in an inconsistent argument position. Since SourceManager is generally not null and it doesn't make sense to apply renaming without one, these inconsistencies are now gone.

### CODE REVIEW

**Review by HerrCai0907 — APPROVED** [review]
LGTM

**Comment by revane** [comment]
I don't have write access. Would someone please land the PR for me?

**Comment by revane** [comment]
@AaronBallman I can't land this PR.

**Pull request merged**
